contracts/contribute/Contributors.sol

```diff
@@ -451,35 +462,27 @@ contract Contributors is ERC721TokenReceiver {
             revert WrongStakingInstance(stakingInstance);
         }
 
-        // Get the token info from the staking contract
-        // If this call fails, it means the staking contract does not have a token and is not compatible
-        address token = IStaking(stakingInstance).stakingToken();
-        // Check the token address
-        if (token != olas) {
-            revert WrongStakingInstance(stakingInstance);
-        }
+        // Get and check service params
+        (address token, uint256 minStakingDeposit, uint32[] memory agentIds, IService.AgentParams[] memory agentParams) =
```

Comment:
The workflow is then consists of 4 parts:
- Get service params;
- Create a new service or update one, if needed;
- Activate and register staking instance;
- Deploy a new service or re-deploy the existing one.